### PR TITLE
[plugin-kafka] Add step for setting headers to event

### PR DESCRIPTION
**Pull request by abudevich**

### CODE REVIEW

File: vividus-tests/src/main/resources/story/system/Kafka.story [hunk 1]
```diff
@@ -11,6 +11,9 @@ Meta:
     @requirementId 2915
 Given I initialize scenario variable `event-value` with value `{"event-from-system-vividus-test": "#{generate(regexify '[a-z]{8}')}"}`
 When I start consuming events from `vividus` Kafka topics `${topic}`
+When I add headers to event:
+|name       |value     |
+|test_header|test_value|
```
Comment (valfirst) [inline vividus-tests/src/main/resources/story/system/Kafka.story:16]: how to verify these headers are sent?

File: vividus-plugin-kafka/src/main/java/org/vividus/steps/kafka/KafkaSteps.java [hunk 2]
```diff
@@ -114,6 +123,25 @@ private <T> Map<String, T> convert(String propertiesPrefix, IPropertyParser prop
                                           factoryCreator::apply))));
     }
 
+    /**
+     * Add kafka event headers
+     * @param headers ExamplesTable representing list of headers with columns "name" and "value" specifying kafka header
+     * names and values respectively
+     */
+    @When("I add headers to event:$headers")
```
Comment (valfirst) [inline vividus-plugin-kafka/src/main/java/org/vividus/steps/kafka/KafkaSteps.java:131]:
```suggestion
    @When("I set Kafka event headers:$headers")
```

File: vividus-plugin-kafka/src/main/java/org/vividus/steps/kafka/KafkaSteps.java [hunk 3]
```diff
@@ -150,7 +178,9 @@ public void sendEvent(String value, String producerKey, String topic)
     public void sendEventWithKey(String key, String value, String producerKey, String topic)
             throws InterruptedException, ExecutionException, TimeoutException
     {
-        kafkaTemplates.get(producerKey).send(topic, key, value).get(WAIT_TIMEOUT_IN_MINUTES, TimeUnit.MINUTES);
+        ProducerRecord record = new ProducerRecord(topic, null, null, key, value,
+                testContext.get(HEADERS_KEY));
```
Comment (valfirst) [inline vividus-plugin-kafka/src/main/java/org/vividus/steps/kafka/KafkaSteps.java:182]: headers must be cleaned up

File: vividus-tests/src/main/resources/story/system/Kafka.story (hunk 1 above)
Comment (abudevich) [reply vividus-tests/src/main/resources/story/system/Kafka.story:16]: discussed, done in KafkaStepsIntegrationTests

File: vividus-plugin-kafka/src/main/java/org/vividus/steps/kafka/KafkaSteps.java [hunk 4]
```diff
@@ -250,10 +282,13 @@ public void processKafkaEvents(QueueOperation queueOperation, String consumerKey
...
-                                                         .map(key -> key == null ? "<no key>" : key)
-                                                         .collect(Collectors.joining(", ")))
-                       .log("Saving events with the keys: {}");
+        LOGGER.atInfo().addArgument(() -> events.stream().map(e -> {
+            String key = e.key() == null ? "<no key>" : e.key();
+            String headerNames = Stream.of(e.headers().toArray()).map(Header::key)
+                    .collect(Collectors.joining(", "));
+            String headersNamesForLog = headerNames.isEmpty() ? "<no headers>" : headerNames;
+            return "{" + key + "; " + headersNamesForLog + "}";
+        }).toList()).log("Saving events with the keys-headers: {}");
```
Comment (valfirst) [inline vividus-plugin-kafka/src/main/java/org/vividus/steps/kafka/KafkaSteps.java:291]:
```suggestion
        LOGGER.atInfo().addArgument(() -> events.stream().map(e -> {
            String key = e.key() == null ? "<no key>" : e.key();
            String headerNames = StreamSupport.stream(e.headers().spliterator(), false)
                    .map(Header::key)
                    .collect(Collectors.joining(", "));
            return headerNames.isEmpty() ? key : "{" + key + "; " + headerNames + "}";
        }).toList()).log("Saving events with the keys and headers: {}");
```

File: vividus-plugin-kafka/src/main/java/org/vividus/steps/kafka/KafkaSteps.java [hunk 5]
```diff
@@ -114,6 +124,25 @@ private <T> Map<String, T> convert(String propertiesPrefix, IPropertyParser prop
                                           factoryCreator::apply))));
     }
 
+    /**
+     * Add kafka event headers
+     * @param headers ExamplesTable representing list of headers with columns "name" and "value" specifying kafka header
+     * names and values respectively
+     */
+    @When("I set Kafka event headers:$headers")
+    public void addEventHeaders(ExamplesTable headers)
```
Comment (valfirst) [inline vividus-plugin-kafka/src/main/java/org/vividus/steps/kafka/KafkaSteps.java:133]:
```suggestion
    public void setEventHeaders(ExamplesTable headers)
```

File: docs/modules/plugins/pages/plugin-kafka.adoc [hunk 6]
```diff
@@ -29,6 +29,30 @@ Where `<producer-key>` is the key of the producer configuration which should be
 
 === Steps
 
+==== *Set headers to Kafka event*
+
+Adds headers to the event, which will be sent to Kafka topic.
```
Comment (valfirst) [inline docs/modules/plugins/pages/plugin-kafka.adoc:34]:
```suggestion
Sets headers to the event, which will be sent to Kafka topic.
```

File: docs/modules/plugins/pages/plugin-kafka.adoc [hunk 7]
```diff
@@ -29,6 +29,30 @@ Where `<producer-key>` is the key of the producer configuration which should be
...
+
+NOTE: The added headers are scoped to the first event, which will be sent, and not available afterwards.
```
Comment (valfirst) [inline docs/modules/plugins/pages/plugin-kafka.adoc:36]:
```suggestion
NOTE: The set headers are scoped to the first event, which will be sent, and not available afterwards.
```

File: docs/modules/plugins/pages/plugin-kafka.adoc [hunk 8]
```diff
@@ -29,6 +29,30 @@ Where `<producer-key>` is the key of the producer configuration which should be
...
+[source,gherkin]
+----
+When I set Kafka event headers:$headers
+----
+
+* `headers` - The xref:ROOT:glossary.adoc#_examplestable[ExamplesTable] representing the list of the headers with columns `name` and `value`.
+
+=== Examples
+
+.Add event header with name `aggregate-type` and value `tenant`
```
Comment (valfirst) [inline docs/modules/plugins/pages/plugin-kafka.adoc:47]:
```suggestion
.Set event header with name `aggregate-type` and value `tenant`
```

**Review by valfirst — CHANGES_REQUESTED** [review]

**Review by valfirst — APPROVED** [review]

**Pull request merged**